### PR TITLE
Clean BuildFiles in DataFormats

### PR DESCRIPTION
**Pull request by guitargeek**
#### PR description:

Another quick partially automatic BuildFile cleaning PR in the style of many before (for example https://github.com/cms-sw/cmssw/pull/31445).

As always, only the dependencies which are **not included in any of the sources** are removed, so these changes are orthogonal and complementary to the recent PRs which added all packages that are **actually included**.

#### PR validation:

CMSSW compiles. It was checked that all newly added dependencies are actually required by the package with `git grep`.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@guitargeek Is the empty file `FWCore/SOA/BuildFile.xml` really needed?

**Comment by guitargeek** [comment]
Yes, it is needed. Recently, @davidlange6 added `FWCore/SOA` as a requirement in a BuildFile:
https://github.com/cms-sw/cmssw/blob/d65a8b6369aa1dbccf71171c08a8b50b5afdc5e0/CommonTools/Utils/BuildFile.xml#L5
This has caused a Warning about including a library that doesn't exist, so we need a BuildFile for FWCore/SOA I guess.

However, SOA has no depedencies other than the standard library, and also doesn't need an export section because it's a header-only library. That's why the BuildFile is empty.

If three is another solution I'm happy to implement it!

**Comment by Dr15Jones** [comment]
@smuzaffar how should we handle a header only CMSSW package?

**Comment by davidlange6** [comment]
Perhaps a time to test the header-only syntax in BuildFiles?
<use name=“FWCore/SOA" source_only="1”/>
Then perhaps that buildfile is not needed?

**Comment by smuzaffar** [comment]
header only packages should also have BuildFile.xml where one can declare its dependencies . For `FWCore/SOA` as there is no external dependencies so there is no need to have `FWCore/SOA/BuildFile.xml` , just adding ` <use name=“FWCore/SOA" source_only="1”/>` in the client's BuildFile should avoid the warnings.

**Comment by guitargeek** [comment]
Ok, thanks, I'll do that!

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by makortel** [comment]
+1

**Comment by agrohsje** [comment]
+1

**Comment by santocch** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by Martin-Grunewald** [comment]
+1

**Comment by perrotta** [comment]
+1
- Technical
- Jenkins tests pass

**Comment by mariadalfonso** [comment]
+xpog

**Comment by qliphy** [comment]
+1

File: DataFormats/ParticleFlowReco/BuildFile.xml [hunk 1]
```diff
@@ -1,7 +1,5 @@
 <use name="DataFormats/CaloRecHit"/>
 <use name="DataFormats/Common"/>
-<use name="DataFormats/EgammaCandidates "/>
```
Comment (davidlange6) [inline DataFormats/ParticleFlowReco/BuildFile.xml:3]: hi @guitargeek - I don't understand this change. DataFormats/EgammaCandidates is used in 3 headers (all of which are then needed to build srcs in this package.

Comment (guitargeek) [reply DataFormats/ParticleFlowReco/BuildFile.xml:3]: Hello! My script is grepping for the exact dependency name, which was here "DataFormats/EgammaCandidates " with a whitespace in the end. By eye this whitespace is not so easy to see when scrolling through the diffs, so I was not prepared for this problem.

I'll make sure to strip the witespaces before grepping next time!

Comment (davidlange6) [reply DataFormats/ParticleFlowReco/BuildFile.xml:3]: ah..I also missed the whitespace.. that explains it:) Thanks.

File: CommonTools/Utils/BuildFile.xml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <use name="DataFormats/Math"/>
 <use name="FWCore/ParameterSet"/>
 <use name="FWCore/Reflection"/>
-<use name="FWCore/SOA"/>
+<use name="FWCore/SOA" source_only="1"/>
```
Comment (makortel) [inline CommonTools/Utils/BuildFile.xml:5]: @smuzaffar I get the following complaint from scram in CMSSW_11_2_X_2020-09-18-1100
```
****WARNING: Invalid tool FWCore/SOA. Please fix src/CommonTools/Utils/BuildFile.xml file.
```
Are we missing something more?

Comment (smuzaffar) [reply CommonTools/Utils/BuildFile.xml:5]: Do you get this in latest IB?

Comment (makortel) [reply CommonTools/Utils/BuildFile.xml:5]: No, in CMSSW_11_2_X_2020-09-22-1100 there is no such complaint. Thanks!

**Pull request merged**